docs/resources/cluster.md

```diff
@@ -121,6 +121,7 @@ resource "argocd_cluster" "eks" {
 * `namespaces` - (Optional) Holds list of namespaces which are accessible in that cluster. Cluster level resources would be ignored if namespace list is not empty..
 * `config` - (Optional) The configuration specification, nested attributes are documented below.
 * `metadata` - (Optional) Cluster metadata, nested attributes are documented below.
+* `project` - (Optional) Scope cluster to ArgoCD project. If omitted, cluster will be global.
```

Comment:
Maybe add a version requirement comment here, and what happens if the field is added on argocd versions < 2.2

Reply:
2.2.0 being required has been added. Also, based on how the application resource does it I added a new feature to the constraints map so if the setting is used with an older version it errors out.